core/modules/user/user.module

```diff
@@ -1772,6 +1789,42 @@ function user_page_title($account) {
   return is_object($account) ? user_format_name($account) : '';
 }
 
+/**
+ * Implements hook_init().
+ */
+function user_init() {
+  $current_path = current_path();
+  $user_login_appearance_setting = user_login_appearance_setting();
+  if (!user_is_logged_in() && ($current_path == 'user/login' || $current_path == 'user/password' || $current_path == 'user/register')) {
```

Comment:
This check `$current_path == 'user/login' || $current_path == 'user/password' || $current_path == 'user/register'` seems to happen several times in this code change.

I wonder, if it makes sense to either make these paths a constant or do the check a reusable callback.

Reply:
Sorted.